### ISSUE TITLE
getStyles() starts returning an empty string when bundled by webpack

### ISSUE THREAD

**Issue by bbrooks** [typestyle/typestyle#216]
tldr; - Typestyle's global `getStyles()` starts returning an empty string serverside when the code is bundled by Webpack, but returns expected styles when transpiled with the typescript compiler (tsc).

Hey there! I'm working on an isomorphic React app that uses typestyle and so far I have really enjoyed the library, nice work!

Typestyle was working well for me until I started compiling my express server with Webpack instead of the typescript compiler. After compiling with Webpack, `getStyles` returns an empty string when I'm sending a response to the client, even though I have declared styles using `style()`.

Here is a simplified picture of a react component that my server renders and returns to the client:

```jsx
const cssPlaceholder = '__CSS_PLACEHOLDER__';
class PageTemplate extends React.Component {
    public render() {
        return (
            <html>
                <head>
                    <style>
                        { cssPlaceholder }
                    </style>
                </head>
                <body>
                    <InnerComponent /> // this component adds styles via the global `style()` method
                </body>
            </html>
        );
    }
}
```

And here is the actual rendering code:
```js
function RenderPage(props) {
    const component = React.createElement(PageTemplate, props);
    const html = renderToStaticMarkup(component);
    return html.replace(cssPlaceholder, getStyles());
}
```

When calling `RenderPage` after compiling with Webpack, `getStyles()` unexpectedly returns an empty string. It returns a list of styles if I compile the code with the typescript compiler. 

After many hours of debugging it seems like the global TypeStyle object `ts` is getting clobbered and reassigned to a `new TypeStyle` at several points during the render process, which would wipe out any styles that my components add along the way. That seems like weird behavior and am wondering if you've seen this before and have any thoughts on why it might be happening just for webpack bundles?

The only fix I can think of is to create my own TypeStyle object and pass it around to my components instead of relying on the global typestyle object. That will be a pretty intensive fix though and I would much rather figure out if I'm using typestyle incorrectly.

Thanks for any help and I'm happy to answer clarifying questions.

Once again, nice work on the library :)

**Comment by notoriousb1t** [comment]
Can you provide a sample of InnerComponent?  I am wondering if it has to do with how the code is declared. When dealing with SSR, there is a big different between doing this:

A.
```ts
const myClass = style({})
export class MyComponent extends React.Component {
    render() {
        return (<div className={myClass}></div>)
    }
}
```

and this:

B.
``` ts
export class MyComponent extends React.Component {
    myClass = style({})
    render() {
        return (<div className={this.myClass}></div>)
    }
}
```

**Comment by bbrooks** [comment]
@notoriousb1t `InnerComponent` follows pattern A, where css classes are defined outside of the react component. Could you explain what the difference is here in terms of SSR?

**Comment by basarat** [comment]
> global TypeStyle object ts is getting clobbered and reassigned to a new TypeStyle at several points during the render process, which would wipe out any styles that my components add along the way. That seems like weird behavior and am wondering if you've seen this before

No I haven't seen this. Place a breakpoint here : https://github.com/typestyle/typestyle/blob/a34a05aa71e61731bcbea43590fc9030bc56e908/src/internal/typestyle.ts#L41 and see what's happening :rose: 

> Could you explain what the difference is here in terms of SSR

The difference would be that `A` only calls `style` once whereas `B` calls `style` for each instance of the component `MyComponent` that gets rendered.  I prefer `A` : https://typestyle.github.io/#/core/tip-code-organization and that is what you have as well. That said, neither should cause any issues (just a small extra function calls that end up being NOOPs performance hit in B)

**Comment by bbrooks** [comment]
@basarat Thanks for the reply. When I have time I'll try to reproduce this in a stripped down implementation that I can share.

**Comment by bbrooks** [comment]
I tried reproducing this with a simple express+typescript+webpack project but the styles came through fine. Closing issue due to non-reproduce-ability.
